From f1ddc1f587e89bab41263ad9b71207da8e16b6ff Mon Sep 17 00:00:00 2001
From: q66 <q66@chimera-linux.org>
Date: Sun, 4 Jun 2023 00:21:55 +0200
Subject: [PATCH] sh(1): make smark/smark2 static

This prevents the compiler from not considering changes done to
the members from within the setjmp conditional.
---
 src.freebsd/sh/main.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src.freebsd/sh/main.c b/src.freebsd/sh/main.c
index 4d1a0e5..955b2bb 100644
--- a/src.freebsd/sh/main.c
+++ b/src.freebsd/sh/main.c
@@ -97,10 +97,11 @@ static char *find_dot_file(char *);
  * is used to figure out how far we had gotten.
  */
 
+static struct stackmark smark, smark2;
+
 int
 main(int argc, char *argv[])
 {
-	struct stackmark smark = {0}, smark2;
 	volatile int state;
 	char *shinit;
 
-- 
2.41.0

